Add docs for 3dequalizer (#245)
ynput/ayon-documentation: website/docs/addon_3dequalizer_admin.md

---
id: addon_3dequalizer_admin
title: 3DEqualizer Admin Docs
sidebar_label: 3DEqualizer
description: AYON 3DEqualizer Addon's documentations for admins.
toc_max_heading_level: 5
---

import ReactMarkdown from "react-markdown";
import versions from '@site/docs/assets/json/Ayon_addons_version.json'

<ReactMarkdown>
{versions.Equalizer_Badge}
</ReactMarkdown>

## Overview

> To get the 3DEqualizer (3DE) addon running, we need to install Qt Python bindings. We use a pre-launch hook that automatically installs `PySide2/6` into 3DE Python. You can set the location using the `TDE4_HOME` environment variable, or it will be deduced from the 3DE executable path in AYON applications settings.

Author: Hey, should that be `TDE4_ROOT` instead?
reference: [Environment variables in 3DE4](https://www.3dequalizer.com/?site=tech_docs&id=121221_01)
```suggestion
To get the 3DEqualizer (3DE) addon running, we need to install Qt Python bindings. We use a pre-launch hook that automatically installs `PySide2/6` into 3DE Python. You can set the location using the `TDE4_ROOT` environment variable, or it will be deduced from the 3DE executable path in AYON applications settings.
```

Reviewer: @antirotor can answer as I dont have the 3DE

<ReactMarkdown>
{versions.Equalizer_Badge}
</ReactMarkdown>

## Overview

> To get 3DEqualizer (3DE) addon running, we need to instal Qt Python bindings - for that we are using pre-launch hook that is automatically installing PySide2/6 into (3DE) Python. You can use ```TDE4_HOME``` environment variable to set the location, or it will be detected from the AYON settings.

Reviewer: ```suggestion
To get 3DEqualizer (3DE) addon running, we need to install Qt Python bindings - for that we are using pre-launch hook that is automatically installing PySide2/6 into (3DE) Python. You can use ```TDE4_HOME``` environment variable to set the location, or it will be detected from the AYON settings.
```

> :::warning Python Bindings

Reviewer: Consider the suggestion below.
```suggestion
:::warning Python Bindings not working for 3DE which installed in protected location.
```

Author: Thank you. it should be resolved in https://github.com/ynput/ayon-documentation/pull/245/commits/54391d84509d492944b555a2dd5e4501a1d60b72

This automatic **PySide2/6** installation might fail if 3DE is used from protected location (either network drive or some user write restricted location). If this happens, you need to install PySide there manually.
:::

### Environment

3DE Addon is using ```AYON_TDE4_HEARTBEAT_INTERVAL``` environment variable to control how often it should yield control from Qt processes back to 3DE. This value can influence stability of 3DE and its responsivness. Default value is 500.

> AYON is also using ```PYTHON_CUSTOM_SCRIPTS_3DE4``` environment variable to install itself to 3DE menu. It is appending to it but if you overwrite this variable using some startup script or rez, AYON menu won't be shown in 3DE.

Reviewer: ```suggestion
AYON is also using ```PYTHON_CUSTOM_SCRIPTS_3DE4``` environment variable to install itself to 3DE menu. This variable appends to AYON but if you overwrite the variable using some startup script or rez, AYON menu won't be shown in 3DE.
```

Author: Thank you. it should be resolved in https://github.com/ynput/ayon-documentation/pull/245/commits/54391d84509d492944b555a2dd5e4501a1d60b72
